kitsune/users/decorators.py

```diff
@@ -0,0 +1,38 @@
+from functools import wraps
+
+from django.contrib.auth.models import User
+from django.http import Http404
+from django.shortcuts import get_object_or_404, redirect
+
+from kitsune.sumo.urlresolvers import reverse
+from kitsune.users.models import Profile
+
+
+def get_profile_from_url(func):
+    @wraps(func)
+    def wrapped_func(request, *args, **kwargs):
+        # The browser replaces '+' in URL's with ' ' but since we never have ' ' in
+        # URL's we can assume everytime we see ' ' it was a '+' that was replaced.
+        # We do this to deal with legacy usernames that have a '+' in them.
+
+        username = kwargs.pop("username")
+        username = username.replace(" ", "+")
+
+        user = User.objects.filter(username=username).first()
```

Comment:
It looks like this is doing two different things. Either redirects to a specific profile or adds the profile in the kwargs. Can we rethink this so that it will do only something specific? I am a bit hesitant on the idea of having a decorator doing redirects

Reply:
I'll remove this change from this PR so we can merge the ES 2.x removal, and file a new issue for this. The redirect it's doing is rather clever (allowing us to use IDs in user routes, rather than usernames which are subject to change), however I'm not sure if we actually make any use of this?

Reply:
https://github.com/mozilla/sumo-project/issues/812